apps/directory/src/components/popovers/NegotiatorSelection.vue

```diff
@@ -139,9 +147,15 @@ export default {
     sortedAlphabetically(collectionArray) {
       return sortCollectionsByLabel(collectionArray);
     },
-    sendRequest() {
-      this.cartVisible = false;
-      this.checkoutStore.sendToNegotiator();
+    async sendRequest() {
+      this.errorMessage = ""; // Reset error message before request
+      try {
+        await this.checkoutStore.sendToNegotiator();
+        this.cartVisible = false;
+      } catch (err) {
+        this.errorMessage =
+          err.message || "An error occurred. Please try again later.";
```

Comment:
Exposing a backend error is not useful for the end user and can break the UI. So always use the generic error message. Logging the actual error to the console is okay.